internal/collector/daemonset.go

```diff
@@ -24,7 +23,7 @@ func (f *Factory) ReconcileDaemonset(er record.EventRecorder, k8sClient client.C
 
 	var receiverInputs []string
 	for _, input := range f.ForwarderSpec.Inputs {
-		if logging.IsHttpReceiver(&input) || logging.IsSyslogReceiver(&input) {
+		if input.Receiver.IsHttpReceiver() || input.Receiver.IsSyslogReceiver() {
```

Comment:
here, we only have 2 kinds of receivers and you must specify a non nil receiver and type or atleast the sub spec, do we really need the function check too.  We can enforce spec requirement in our internal/validators.... if receiver is non nil then you must specify one of the other

Reply:
That would check a different thing. Here, we want to make sure that the Receiver is http or syslog. Other receiver types might do something different. I feel comfortable with this specificity.

Reply:
If we would like to be less specific, this could be replaced with `input.Receiver != nil` because our current validators+migration combination make sure that whether we have a Receiver, we will have a receiverType. I prefer the way it is now, though.

Reply:
I think its reasonable to only check for nil.  I presume if we implement another receiver type the behavior would be the same: we want to identify it as a receiver input